### PR TITLE
[AutoParallel] Add benchmark for llama-7b-dy2st.

### PR DESCRIPTION
**Pull request by GhostScreaming**
### PR types
Others

### PR changes
Others

### Description
Add llama2-7b for test_tipc

### CODE REVIEW

File: tests/test_tipc/static/auto_parallel/llama2/pretrain_config_llama2_7b/pretrain-llama2_7b.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  "model_name_or_path": "meta-llama/Llama-2-7b",
+  "tokenizer_name_or_path": "meta-llama/Llama-2-7b",
+  "input_dir": "./data",
+  "output_dir": "./checkpoints/llama2_pretrain_ckpts",
+  "per_device_train_batch_size": 1,
+  "gradient_accumulation_steps": 1,
+  "per_device_eval_batch_size": 2,
+  "tensor_parallel_degree": 1,
+  "pipeline_parallel_degree": 1,
+  "sharding": "stage1",
```
Comment (heavyrain-lzy) [inline tests/test_tipc/static/auto_parallel/llama2/pretrain_config_llama2_7b/pretrain-llama2_7b.json]: 动态图`stage1`跟静态图`stage2`一致。

File: tests/test_tipc/static/auto_parallel/llama2/pretrain_config_llama2_7b/pretrain-llama2_7b.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  "sharding_parallel_config": "enable_stage1_overlap",
```
Comment (heavyrain-lzy) [inline tests/test_tipc/static/auto_parallel/llama2/pretrain_config_llama2_7b/pretrain-llama2_7b.json]: "enable_stage1_overlap" --> "enable_stage2_overlap"

File: tests/test_tipc/static/auto_parallel/llama2/pretrain_config_llama2_7b/pretrain-llama2_7b.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  "tensor_parallel_config": "enable_delay_scale_loss enable_mp_async_allreduce enable_mp_skip_c_identity enable_mp_fused_linear_param_grad_add",
```
Comment (heavyrain-lzy) [inline tests/test_tipc/static/auto_parallel/llama2/pretrain_config_llama2_7b/pretrain-llama2_7b.json]: "tensor_parallel_config": "enable_delay_scale_loss enable_mp_async_allreduce enable_mp_skip_c_identity enable_mp_fused_linear_param_grad_add",
-->
"tensor_parallel_config": "enable_mp_async_allreduce",

File: tests/test_tipc/static/auto_parallel/llama2/pretrain_config_llama2_7b/pretrain-llama2_7b.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  "pipeline_parallel_config": "enable_delay_scale_loss enable_release_grads disable_partial_send_recv",
```
Comment (heavyrain-lzy) [inline tests/test_tipc/static/auto_parallel/llama2/pretrain_config_llama2_7b/pretrain-llama2_7b.json]: 没有开PP并行，"pipeline_parallel_config": ""设置为空即可

File: tests/test_tipc/static/auto_parallel/llama2/pretrain_config_llama2_7b/pretrain-llama2_7b.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  "min_learning_rate": 3e-06,
+  "warmup_steps": 30,
+  "logging_steps": 1,
+  "max_steps": 50,
+  "save_steps": 5000,
+  "eval_steps": 1000,
+  "weight_decay": 0.01,
+  "bf16": true,
+  "fp16_opt_level": "O2",
+  "amp_master_grad": true,
```
Comment (heavyrain-lzy) [inline tests/test_tipc/static/auto_parallel/llama2/pretrain_config_llama2_7b/pretrain-llama2_7b.json:34]: 增加：
"amp_custom_black_list": ["reduce_sum", "c_softmax_with_cross_entropy"],
  "amp_custom_white_list": ["lookup_table", "lookup_table_v2"],

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**